app/assets/javascripts/discourse/app/helpers/share-url.js

```diff
@@ -0,0 +1,8 @@
+import { helperContext } from "discourse-common/lib/helpers";
+
+export function resolveShareUrl(url, user) {
+  const badgesEnabled = helperContext().siteSettings.enable_badges;
```

Comment:
Feels like reaching out too far. Why don't you pass the `enable_badges` site setting value to the function instead? It should be available from the callers.

Reply:
Sure, I can do that, but I feel like if we do this then the helper is redundant because it would only return one value based on a three parameters that we pass to it from the callers. 

```
const userSuffix = user && badgesEnabled ? `?u=${user.username_lower}` : "";

return url + userSuffix;
```

And we would need to duplicate some code in both callers (there's only two for now)

Let me know what works for you.

Reply:
My threshold for refactoring "duplicated" codes into a method is 3, so not quite 2 ;)

Up to you really. I'm fine either way (especially since you've already done most of the work).